services/webhook/deliver.go

```diff
@@ -97,6 +97,7 @@ func Deliver(t *models.HookTask) error {
 	req.Header.Add("X-Gogs-Delivery", t.UUID)
 	req.Header.Add("X-Gogs-Event", t.EventType.Event())
 	req.Header.Add("X-Gogs-Signature", t.Signature)
+	req.Header.Add("X-Hub-Signature", t.SignatureGithub)
```

Comment:
Why doesn't this work?

```suggestion
	req.Header.Add("X-Hub-Signature-256", "sha256="+t.Signature)
```

Reply:
@zeripath 

I tested that on my side it doesn't work for me using Ansible AWX as the upstream integration point (https://github.com/ansible/awx/blob/devel/awx/api/views/webhooks.py).  It expects X-Hub-Signature only with sha1. It doesn't accept X-Hub-Signature-256.

`logger.debug("Expected signature missing from header key HTTP_X_HUB_SIGNATURE")`

Not sure if 'X-Hub-Signature-256' is universally supported out there. If there is a reason not to support X-Hub-Signature with sha1 in gitea, I can change the PR to do X-Hub-Signature-256 only and open an issue with Ansible AWX project.

Reply:
Ok. If you're certain that the header I suggest above isn't supported I'd suggest we still add it but... in addition we should adjust your PR:

* SignatureGitHub should become SignatureSHA1
* Don't store it with the `sha1=` prefix.
* Only add the prefix when setting the header.

If we're pre calculating this we're going to need to write a migration to update any stored webhook to add the sha1 signature to the db. Otherwise we'll have to have some logic that tests if the Sha1 signature is empty before emitting the header.

Reply:
@zeripath
it makes perfect sense. I put in those changes but haven't pushed just yet. In terms of migration, I took a look at it. I might be able to add a migration, I just don't know how to test it properly. Does the migration run on a certain condition or every time?